src/kernels/jupyter/types.ts

```diff
@@ -189,11 +189,12 @@ export interface IJupyterServerUri {
 
 export type JupyterServerUriHandle = string;
 
-export interface IJupyterUriProvider {
+export interface IInternalJupyterUriProvider {
```

Comment:
This type is used internally and the only difference between this and the exported type for API is this has `extensionId`
Similar to VS Code, we have two types, one for exported APIs and one for internal.
This way we can add `extensionId` to the internal type. else it gets too messy,  other alternatives is always welcome